packages/terra-application-layout/examples/index-examples/ExampleApplication.jsx

```diff
@@ -0,0 +1,326 @@
+      title: 'Additional Item 1 - Sub 2',
+      parentKey: 'additional-1',
+    }, {
+      key: 'checkbox-item',
+      title: 'Custom Checkbox Item',
+      isSelectable: true,
+      isSelected: checkboxItemEnabled,
+      parentKey: Utils.utilityHelpers.defaultKeys.MENU,
+    }, {
+      key: 'additional-3',
+      contentLocation: Utils.utilityHelpers.locations.FOOTER,
+      title: 'Custom Footer',
+      parentKey: Utils.utilityHelpers.defaultKeys.MENU,
+    }];
+
+    /**
+     * The data provided for utilityConfig will be visible in the ApplicationLayout's header in the
+     * standard rendering mode and within the menus in the compact rendering mode.
+     *
+     * The ApplicationLayout's Utils export provides a helper function named getDefaultUtilityConfig that will
+     * generate the configuration for the standard set of utility options. If the standard configuration is not
+     * desirable, an entirely custom configuration can be used instead.
+     */
+    const utilityConfig = Object.freeze({
+      title: 'Swanson, Henry',
+      accessory: userAvatar,
+      menuItems: Utils.utilityHelpers.getDefaultUtilityItems(intl, userData, customUtilityItems),
+      initialSelectedKey: Utils.utilityHelpers.defaultKeys.MENU,
+      onChange: (event, itemData, disclose) => {
+        /**
+         * This function will be called when items are selected within the utility menu.
+         * The disclose parameter is provided for convenience, but any presentation method
+         * could be used to handle that menu content selection.
+         */
+
+        /**
+         * For the checkbox item, we maintain its selection state locally and show no disclosure.
+         */
+        if (itemData.key === 'checkbox-item') {
+          this.setState({
+            checkboxItemEnabled: !checkboxItemEnabled,
+          });
+          return;
+        }
+
+        /**
+         * For everything else, we can present a custom modal component for that content.
+         */
+        disclose({
+          preferredType: 'modal',
+          size: 'small',
+          content: {
+            key: itemData.key,
+            component: <UtilityOption name={itemData.key} />,
+          },
+        });
+      },
+    });
+
+    return (
+      <ContentContainer
+        fill
+        header={<h3>{`Router Location: ${location.pathname}`}</h3>}
+      >
+        <ApplicationLayout
+          nameConfig={nameConfig}
+          utilityConfig={utilityConfig}
+          routingConfig={routingConfig}
+          navigationItems={navigationItems}
+          extensions={<ApplicationExtensions />}
+          indexPath={indexPath}
+        />
+      </ContentContainer>
+    );
+  }
+}
+
+ExampleApplication.propTypes = {
+  intl: intlShape,
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }),
+};
+
+const WrappedApplication = withRouter(injectIntl((ExampleApplication)));
+
+const AppRouter = () => (
+  <div style={{ height: '100%' }}>
```

Comment:
This highlights an issue that isn't immediately transparent that we've talked about before. For the layout to render as expected, it's parent containers must be set to have `height:100%;` Using the following config:
```
<BrowserRouter>
  <Base locale="en-US">
    <ApplicationLayout
      indexPath={indexPath}
      nameConfig={nameConfig}
      navigationItems={navigationItems}
      routingConfig={routingConfig}
    />
  </Base>
</BrowserRouter>
```
Results in a layout that is only as tall as the header.
![menu](https://user-images.githubusercontent.com/633148/37785821-1e11b846-2dc9-11e8-9247-6bab63117635.gif)

Looking at the generated markup, both the div#root and the direct child div rendered by the `<Base />` component need `height: 100%` set. 

<img width="429" alt="screen shot 2018-03-22 at 12 05 32 pm" src="https://user-images.githubusercontent.com/633148/37785998-8fba99ae-2dc9-11e8-9ad1-3ca5deb37600.png">

I've logged #124 to dive into this some more but it doesn't need to hold up this code review.